rholang/src/main/scala/coop/rchain/rholang/interpreter/Interpreter.scala

```diff
@@ -68,26 +76,38 @@ object Interpreter {
-                          phlosLeft <- C.get
-                          oldErrors <- errorLog.readAndClearErrorVector()
-                          newErrors = result.swap.toSeq.toVector
-                          allErrors = oldErrors |+| newErrors
-                        } yield EvaluateResult(initialPhlo - phlosLeft, allErrors)
-                      case Left(error) =>
-                        EvaluateResult(parsingCost, Vector(error)).pure[F]
-                    }
-                  case Left(error) =>
-                    EvaluateResult(parsingCost, Vector(error)).pure[F]
-                }
-        } yield res
-
+          evaluateResult <- parseResult match {
+                             case Right(_) =>
+                               ParBuilder[F].buildNormalizedTerm(term, normalizerEnv).attempt >>= {
+                                 case Right(parsed) =>
+                                   for {
+                                     injResultEither   <- reducer.inj(parsed).attempt
+                                     phlosLeft         <- C.get
+                                     interpreterErrors <- errorLog.readAndClearErrorVector()
+                                     evaluateResult <- injResultEither.swap.toOption match {
+                                                        case Some(OutOfPhlogistonsError) =>
+                                                          EvaluateResult(
+                                                            initialPhlo - phlosLeft,
+                                                            OutOfPhlogistonsError +: interpreterErrors
+                                                          ).pure[F]
+                                                        case Some(throwable) =>
+                                                          throwable.raiseError[F, EvaluateResult]
+                                                        case None =>
+                                                          EvaluateResult(
+                                                            initialPhlo - phlosLeft,
+                                                            interpreterErrors
+                                                          ).pure[F]
+                                                      }
+                                   } yield evaluateResult
+                                 case Left(interpreterError: InterpreterError) =>
+                                   EvaluateResult(parsingCost, Vector(interpreterError)).pure[F]
+                                 case Left(throwable) => throwable.raiseError[F, EvaluateResult]
+                               }
+                             case Left(interpreterError: InterpreterError) =>
+                               EvaluateResult(parsingCost, Vector(interpreterError)).pure[F]
+                             case Left(throwable) => throwable.raiseError[F, EvaluateResult]
```

Comment:
I'm not sure what happened here, but I think I prefer the previous version... :)

Reply:
I'm catching all of the `InterpreterErrors` and returning them in `EvaluateResult` and bubbling all other throwables up to the next level. I don't really see a less verbose way of doing this, and I'm not confident that all errors coming from the normalizer have been caught and converted to `InterpreterError`.